app/client/src/app/components/form-components/autoCompleteWrapper.js

```diff
@@ -22,21 +23,39 @@ export default function AutoCompleteWrapper(props) {
     const startIdx = 0;
     const endIdx = 7;
 
-    return inputLength === 0
-      ? []
-      : props.suggestions
-          .filter(
-            school => school.toLowerCase().slice(0, inputLength) === inputValue
-          )
-          .slice(startIdx, endIdx);
+    // Return no suggestions if the input is empty
+    if (inputLength == 0) {
+      return [];
+    }
+
+    // Construct an array of potential school suggestions.
+    // First check the input against alternative school names,
+    // then, concat that list with the suggestions from the full list of schools.
+    var potentialSchools = Object.keys(alternativeSchoolNames)
+      .filter(school => school.includes(inputValue))
+      .map(alternativeName => alternativeSchoolNames[alternativeName])
+      .concat(
+        // List of all official school names that contact the input value
+        props.suggestions.filter(school =>
+          school.toLowerCase().includes(inputValue)
+        )
+      );
+
+    // Remove duplicates from the potential schools array
+    // and slice the array down to a smaller size
+    potentialSchools = potentialSchools
+      .filter((school, idx) => idx == potentialSchools.indexOf(school))
+      .slice(startIdx, endIdx);
+
+    return potentialSchools;
   }
 
   function getSuggestionValue(suggestion) {
     props.autoCompleteSelect(suggestion);
   }
 
-  function onSuggestionsFetchRequested() {
-    return setSuggestions(getSuggestions(props.value));
+  function onSuggestionsFetchRequested({ value }) {
+    return setSuggestions(getSuggestions(value));
```

Comment:
So apparently `props.value` gets updated _after_ `onSuggestionsFetchRequested` is called. Therefore the suggestions returned are always one character behind (e.g. "uci" returns suggestions for "uc").

When `onSuggestionsFecthRequested` is called, it actually passes through the current value of the input field, so I just use that here instead of `props.value`.

Reply:
nice!